experimental/oak_baremetal_runtime/src/framing.rs

```diff
@@ -25,69 +28,115 @@ use oak_remote_attestation::handshaker::{
 };
 use oak_remote_attestation_sessions::SessionId;
 
-struct InvocationHandler<F, G, V>
+enum InitializationState<G, V>
+where
+    G: AttestationGenerator,
+    V: AttestationVerifier,
+{
+    Uninitialized(Option<AttestationBehavior<G, V>>),
+    Initialized(Box<dyn AttestationHandler>),
```

Comment:
Drive-by comment: can this not be replaced by a generic paramater? (e.g. type parameter `H` where `H: AttestationHandler` and then either `Initialized(Box<H>)` or even just `Initialized<H>`)

Reply:
It could be yes `Initialized(Box<H>)`, or `Initialized<H>`, if we require `H` to be Sized. :) We'd still have to implement scope the type down in the implementation of the `InvocationHandler`, since it invokes methods of the `AttestationHandler` trait. :)

Reply:
Added a comment to explain why dyn is used. :) It might be possible to remove the need for it down the road, but likely requires refactoring the attestation logic.